app/questionnaire/placeholder_transforms.py

```diff
@@ -139,6 +149,79 @@ def calculate_date_difference(first_date: str, second_date: str) -> str:
         )
         return day_string.format(number_of_days=time.days)
 
+    def generate_date_range(
+        self,
+        reference_date: str,
+        offset_full_weeks: int,
+        days_in_range: int,
+        first_day_of_week: str = "monday",
+    ) -> tuple[str, str]:
+        """Generate a start and end date for a date range given a reference date,
+        weeks prior and number of days in range.
+
+        :param reference_date: The date to reference the date range from, can be YYY&-MM-DD, YY-MM or 'now'.
+        :type reference_date: datetime
+        :param offset_full_weeks: Number of full weeks to offset from the reference date to start the date range.
+        :type offset_full_weeks: int
+        :param days_in_range: Number of days in the range, including the start date, must be a positive integer.
+        :type days_in_range: int
+        :param first_day_of_week: Which day of the week should be considered the first (default 'monday').
+            This is the day of the week which is selected as the start of the range.
+        :type first_day_of_week: str
+        :return: The start and end datetime objects of the range.
+        :rtype: Tuple[datetime, datetime]
+        """
+        reference_datetime = PlaceholderTransforms.parse_date(reference_date)
+        try:
+            first_day_of_week_idx = DAYS[first_day_of_week.lower()]
+        except KeyError as err:
+            raise KeyError(f"'{first_day_of_week}' is not a valid weekday") from err
+
+        if days_in_range < 1:
+            raise ValueError("'days_in_range' must be a positive integer")
+
+        first_day_of_last_partial_week = reference_datetime - timedelta(
```

Comment:
Is this now always the `current_week` of the reference date?. It is not the last week?.

Reply:
Yes it is the current week - current week is the same as last_partial_week. I can rename to clarify